### ISSUE TITLE
Issue in the block "any of ..." in variadic reporter library

### ISSUE THREAD

**Issue by show0k** [jmoenig/Snap--Build-Your-Own-Blocks#1100]
This block is imported with the "variadic reporter" library but seem to not working well.

If I use the inputs with the left or right arrows, it works well.
![any_original1](https://cloud.githubusercontent.com/assets/3891499/12141534/8fbed458-b470-11e5-85d5-3414d1853b26.png)
But if I use an input list, the "call" block try to execute the booelan...
![any_original](https://cloud.githubusercontent.com/assets/3891499/12141574/de69d602-b470-11e5-8067-2b06ccea0694.png)

If I rigify the statement in the "call" block it works well
![any_fix](https://cloud.githubusercontent.com/assets/3891499/12141616/5cf66076-b471-11e5-8754-78e923971b23.png)
Or we can even delete the call block (I don't understand what it was intended to do).

**Comment by cycomachead** [comment]
I ran into this a while ago, and I think I know the issue, but I'd like to hear Brian's thoughts.

The issue, I believe is that `preds` is of type `boolean (unevaluated)`, so it auto-lambda-ifies the booleans input. This is so `any` can short-circuit, I think. What I guess would need to happen, is that if you pass an input list into a variadic uneval'd solt, Snap<em>!</em> should auto-lambda-ify the ITEMS of the list.

**Comment by brianharvey** [comment]
Yes that sounds right to me.  Reassigning to Jens and relabelling. :-)

**Comment by jmoenig** [comment]
I'm not sure about this. My inclination is that Snap is absolutely doing the right thing here, and that if you decide to pass in a list of inputs the caller has to make sure to reify those inputs explicitly, same as if the user passes in a list of other inputs Snap does *not* make sure all the items in that list conform to the specified slot type (or throw an error).

I need to think about this, but my current inclination is to claim that Snap is alright and that unevaluated variadic inputs require careful handling rather than magic gimmicks.

**Comment by brianharvey** [comment]
The thing is, the whole point of unevaluated input types is that the *user* doesn't know that's what they are.  S/he thinks they're just (in this case) Booleans.

**Comment by jmoenig** [comment]
Yeah, I know, that's why I need to think about this... :-)

**Comment by jmoenig** [comment]
One problem is that a common use for the ```WITH INPUT LIST``` case is to recursively use the reporter inside its own definition body, as happens to be the case with the ```ANY OF``` reporter here. In this case *not* auto-lambdafying the elements of the input list is crucial since - for the recursive call - they already are reified (in the "normal" usage).

Basically we could add a type-assertion-cum-automagic-type-casting, which is what you're suggesting. That makes me very uncomfortable, at least on the *language* level. I guess I wouldn't mind if you wrote that

  ```ASSERT ALL ELEMENTS OF lst ARE predicates```

in Snap itself, because then it would be visible to the user...

**Comment by cycomachead** [comment]
> One problem is that a common use for the WITH INPUT LIST case is to recursively use the reporter inside its own definition body, as happens to be the case with the ANY OF reporter here. In this case not auto-lambdafying the elements of the input list is crucial since - for the recursive call - they already are reified (in the "normal" usage).

Yeah, I think this came up in some other discussion we had. One of the things that was discussed would be to have uneval'd inputs _not_ auto-lambda-ify lambdas, so you wouldn't have the recursive problem, but I think that would break other parts of lambda, IIRC. :/

**Comment by brianharvey** [comment]
Let me think about this overnight.  I'm neck-deep in deadlines today.
